src/app/content/reducer.ts

```diff
@@ -74,6 +75,10 @@ function reduceContent(state: State, action: AnyAction) {
     case getType(actions.receivePage): {
       return reduceReceivePage(state, action);
     }
+    case getType(actions.receivePageNotFound): {
+      const loading = omit('page', state.loading);
+      return {...omit(['page', 'references'], state), loading, pageNotFound: action.payload};
```

Comment:
I've changed it to `pageNotFoundId` and left `pageNotFound` selector
I know that in this app it's pretty common patter to call additional actions on `locationChange` but I find it very hard to maintain when we have so many side effects

Reply:
Well it wouldn't need dispatch additional actions, i imagine it would only require you to modify [this line](https://github.com/openstax/rex-web/blob/04108ef9bedbb19a22030ebc676009697e5fe2bb/src/app/content/reducer.ts#L94), unless I'm missing something.

I agree about the side effects making it harder to maintain although am not really sure if this counts as it.

Reply:
I was thinking about dispatching clearPageNotFound on locationChange.
Using existing `locationChange` in the reducer to reset it seems better, however, it would still require us to maintain two places - or actually 5 places (there are 4 `return` statements in `locationChange` reducer) - instead of one place
I think moving this logic to `pageNotFound` selector makes it simpler